molecule/provisioners/libvirtprovisioner.py

```diff
@@ -0,0 +1,923 @@
+#  Copyright (c) 2015-2016 Cisco Systems
```

Comment:
Couple of general remarks so far:
- please add docstrings to all the methods/functions
- you really want to simplify the flow of the methods in here, in particular you want to limit the number of nested blocks; this can be achieved by using a more /functional/ style when appropriate (i.e. always); I suggested a couple of improvements to that effect, in particular the usage of `next`.
- if would be nice to use functions (i.e. not class methods, functions) for little things that can be abstracted away and don't need context (writing/reading files, dealing with templates, etc.).